### DISCUSSION TITLE
Unable to use with typescript due to a compile error

### DISCUSSION THREAD

**Discussion by realtebo** [nodejs/undici#3010] [General]
this simple code

```

import { request } from 'undici';

export async function getDevices() {

  const catResult = await request('https://aws.random.cat/meow');

  return {};
}

```

i am using this with nextjs-14, i don't know if it is relevant

but when running the code I got this

```
Failed to compile
./node_modules/undici/lib/web/fetch/util.js
Module parse failed: Unexpected token (812:57)
|       // 5. If object is not a default iterator object for interface,
|       //    then throw a TypeError.
>       if (typeof this !== 'object' || this === null || !(#target in this)) {
|         throw new TypeError(
|           `'next' called on an object that does not implement interface ${name} Iterator.`
```

**Comment by truexor** [comment]
given it is an async function, you would use `'use server'` on the top level.

Could you elaborate on where the code is, as in the file path. That could help.

**Comment by Alekzv9** [comment]
The same error happening to me with `Using the PlanetScale serverless driver with Prisma` guide

https://planetscale.com/docs/tutorials/planetscale-serverless-driver-prisma-example


"next": "^14.1.4",
    "undici": "^6.12.0",
"@planetscale/database": "^1.16.0",
    "@prisma/adapter-planetscale": "^5.12.1",
    "@prisma/client": "^5.12.1",

**Comment by Uzlopak** [comment]
This is a next issue. Please tackle this with them. 

Duplicate of #2923

**Reply by Alekzv9** [reply]
I see, I downgrade to undici@5.28.4 and it worked

**Comment by titouancreach** [comment]
Just so if people land here:

If `undici` is used as a peer dep, and you have no idea why this is all about, in your package.json add

pnpm:
```json
  "pnpm": {
    "overrides": {
      "undici": "5.28.4"
    }
  }
```

npm:
```json
  "overrides": {
    "undici": "5.28.4"
  }
```

**Comment by nukeop** [comment]
I found this discussion because of this upgrade in @distube/ytdl-core: https://github.com/distubejs/ytdl-core/commit/efa92e91bdc18fd50897d0cb35ed96f4c86cc858

I am not even using this package directly. Next.js was never near my project.

**Comment by sumthedev** [comment]
hey, im also facing this error, please let me know how to fix, i read the above discussion, but its doesnt help
```
`./node_modules/undici/lib/web/fetch/util.js
Module parse failed: Unexpected token (682:63)
|             // 5. If object is not a default iterator object for interface,
|             //    then throw a TypeError.
>             if (typeof this !== "object" || this === null || !(#target in this)) {
|                 throw new TypeError(`'next' called on an object that does not implement interface ${name} Iterator.`);
|             }`
```

**Reply by realtebo** [reply]
sorry, for me, I gave up. I switched to zod.

**Reply by pushkarsingh32** [reply]
> `./node_modules/undici/lib/web/fetch/util.js
> Module parse failed: Unexpected token (682:63)
> |             // 5. If object is not a default iterator object for interface,
> |             //    then throw a TypeError.
> >             if (typeof this !== "object" || this === null || !(#target in this)) {
> |                 throw new TypeError(`'next' called on an object that does not implement interface ${name} Iterator.`);
> |             }`

did you manage to solve it?

**Reply by realtebo** [reply]
Sorry to Say no. I switched to zod
